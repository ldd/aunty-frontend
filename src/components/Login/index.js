import React from "react";

export const BACKEND_URL =
  process.env.NODE_ENV === "production"
    ? "https://auntyapp.io/api"
    : "http://localhost:4000";

const SLACK_CLIENT_ID = "695399984727.687207103297";

const SLACK_REDIRECT_URI = encodeURI(`${BACKEND_URL}/auth/slack`);

export const redirectToLogin = () => {
  // we want to force the page to reload, to the login page
  window.location.href = "/#Login";
  window.location.reload();
};

const Login = () => (
  <div className="hero-body">
    <div className="container has-text-centered">
      <div className="columns">
        <div className="column is-narrow">
          {/* from slack's button generator https://api.slack.com/docs/sign-in-with-slack#generator */}
          <a
            href={`https://slack.com/oauth/authorize?scope=identity.basic&client_id=${SLACK_CLIENT_ID}&redirect_uri=${SLACK_REDIRECT_URI}`}
          >
            <img
              alt="Sign in with Slack"
              height="40"
              width="172"
              src="https://platform.slack-edge.com/img/sign_in_with_slack.png"
              srcSet="https://platform.slack-edge.com/img/sign_in_with_slack.png 1x, https://platform.slack-edge.com/img/sign_in_with_slack@2x.png 2x"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
);

export default Login;
