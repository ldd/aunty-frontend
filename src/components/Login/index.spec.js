import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Login, { redirectToLogin } from ".";

describe("Heading", () => {
  it("renders snapshot", () => {
    const component = renderer.create(<Login />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("redirects to login", () => {
    // mocking window.location.reload
    // https://gist.github.com/remarkablemark/5cb571a13a6635ab89cf2bb47dc004a3
    const { location } = window;
    delete window.location;
    window.location = { reload: jest.fn() };

    expect(window.location.reload).not.toHaveBeenCalled();
    redirectToLogin();
    expect(window.location.reload).toHaveBeenCalled();
  });
});
