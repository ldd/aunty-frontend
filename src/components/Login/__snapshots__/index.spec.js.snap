// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading renders snapshot 1`] = `
<div
  className="hero-body"
>
  <div
    className="container has-text-centered"
  >
    <div
      className="columns"
    >
      <div
        className="column is-narrow"
      >
        <a
          href="https://slack.com/oauth/authorize?scope=identity.basic&client_id=695399984727.687207103297&redirect_uri=http://localhost:4000/auth/slack"
        >
          <img
            alt="Sign in with Slack"
            height="40"
            src="https://platform.slack-edge.com/img/sign_in_with_slack.png"
            srcSet="https://platform.slack-edge.com/img/sign_in_with_slack.png 1x, https://platform.slack-edge.com/img/sign_in_with_slack@2x.png 2x"
            width="172"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
